graylog2-server/src/main/java/org/graylog2/alerts/AlertServiceImpl.java

```diff
@@ -17,122 +17,113 @@
+import org.graylog2.bindings.providers.MongoJackObjectMapperProvider;
+import org.graylog2.database.CollectionName;
 import org.graylog2.database.MongoConnection;
 import org.graylog2.database.NotFoundException;
-import org.graylog2.database.PersistedServiceImpl;
 import org.graylog2.plugin.Tools;
 import org.graylog2.plugin.alarms.AlertCondition;
+import org.graylog2.plugin.database.ValidationException;
 import org.graylog2.plugin.streams.Stream;
 import org.graylog2.rest.models.streams.alerts.requests.CreateConditionRequest;
 import org.joda.time.DateTime;
-import org.joda.time.DateTimeZone;
 import org.joda.time.Seconds;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
+import org.mongojack.DBQuery;
+import org.mongojack.DBSort;
+import org.mongojack.JacksonDBCollection;
 
 import javax.inject.Inject;
 import java.util.List;
-import java.util.Locale;
 import java.util.Map;
+import java.util.stream.Collectors;
 
 import static com.google.common.base.Preconditions.checkArgument;
 
-public class AlertServiceImpl extends PersistedServiceImpl implements AlertService {
-    private static final Logger LOG = LoggerFactory.getLogger(AlertServiceImpl.class);
-
-    private final Map<String, AlertCondition.Factory> alertConditionMap;
+public class AlertServiceImpl implements AlertService {
+    private final JacksonDBCollection<AlertImpl, String> coll;
+    private final AlertConditionFactory alertConditionFactory;
 
     @Inject
     public AlertServiceImpl(MongoConnection mongoConnection,
-                            Map<String, AlertCondition.Factory> alertConditionMap) {
-        super(mongoConnection);
-        this.alertConditionMap = alertConditionMap;
+                            MongoJackObjectMapperProvider mapperProvider,
+                            AlertConditionFactory alertConditionFactory) {
+        this.alertConditionFactory = alertConditionFactory;
+        final String collectionName = AlertImpl.class.getAnnotation(CollectionName.class).value();
+        final DBCollection dbCollection = mongoConnection.getDatabase().getCollection(collectionName);
+        this.coll = JacksonDBCollection.wrap(dbCollection, AlertImpl.class, String.class, mapperProvider.get());
     }
 
     @Override
-    public Alert factory(AlertCondition.CheckResult checkResult) {
-        if (!checkResult.isTriggered()) {
-            throw new RuntimeException("Tried to create alert from not triggered alert condition result.");
-        }
-
-        Map<String, Object> fields = Maps.newHashMap();
-        fields.put("triggered_at", checkResult.getTriggeredAt());
-        fields.put("condition_id", checkResult.getTriggeredCondition().getId());
-        fields.put("stream_id", checkResult.getTriggeredCondition().getStream().getId());
-        fields.put("description", checkResult.getResultDescription());
-        fields.put("condition_parameters", checkResult.getTriggeredCondition().getParameters());
+    public Alert.Builder builder() {
```

Comment:
This is the extension of factory methods which we have in most of our DAO interfaces, following our pattern that value classes are constructed by DAO classes only. If we want to change this (if there are good reasons to do that), then we should do it in a separate PR and consistently.